### ISSUE TITLE
FuriousFPV Innova OSD-VTX support

### ISSUE THREAD

**Issue by jflyper** [ShikOfTheRa/scarab-osd#327]
For discussion and info exchange on support for FuriousFPV's Innova VTX.

The Innova basically seems like M328P driving MAX7456 and FX758-2. A vendor modified (and not yet disclosed) version of MWOSD (of unknown release) is used.

Estimated changes are as follow.
1. Bit-banging SPI-ish controlling FX758-2 through 3 (or 2) digital output pins.
2. Config menu extension which thoughtlessly squeezed 3 more rows for vtx parameters into PID page (and made the product incompatible with NTSC).

Product page:
[INNOVA OSD-VTX board 20x20mm](http://furiousfpv.com/product_info.php?products_id=76)

RCG thread (PIKO BLX):
[PIKO BLX Micro Flight Controller - Everything Changes](https://www.rcgroups.com/forums/showthread.php?t=2664875&page=75#post36053035)

FX758-2:
[https://szfxt.en.alibaba.com/product/60303419485-214503132/FXT_Lightweight_5_8GHz_200mW_VTX_FX758_Support_SPI_AV_Transmitter_Module_for_Raceband_VTX.html](https://szfxt.en.alibaba.com/product/60303419485-214503132/FXT_Lightweight_5_8GHz_200mW_VTX_FX758_Support_SPI_AV_Transmitter_Module_for_Raceband_VTX.html)
[http://www.banggood.com/FPV-5_8G-200mW-32CH-Wireless-Audio-Video-AV-Transmitter-Module-FX758-2-p-980317.html](http://www.banggood.com/FPV-5_8G-200mW-32CH-Wireless-Audio-Video-AV-Transmitter-Module-FX758-2-p-980317.html)

ImpulseRC's RF module is different; looks Aomway style, with RTC6705 and frequency controlling MCU in the shield.

EDIT
Helix's RF module seems to be a one of the SUNFLYTEK SM21xT family, probably a revision of SM215T with 40ch upgraded firmware, or MCU-less version.
[https://shop1396059262556.1688.com/page/offerlist_53381818.htm?showType=windows](https://shop1396059262556.1688.com/page/offerlist_53381818.htm?showType=windows)

**Comment by jflyper** [comment]
Related
#315 Smart audio support
#323 Where is the code for controlling a 6705 5.8ghz video transmitter?

#273 add support for BetaFlight custom MSP commands
#325 Canvas mode support for BF-CMS and inverted char support
betaflight/#1352 Look Ma! BF config menu runs on my Minim!
betaflight/#1283 Add TBS SmartAudio support

#321 Support for Furious FPV Piggy OSD (Not technically related)

**Comment by jflyper** [comment]
Since the bit-banged SPI for MAX7456 is order of several magnitudes easier than non-blocking software serial, I would say it's a weekend job to add a RTC6705 driver, and another to add a menu page, once started. (Adding a menu page would take more time... sigh.)

**Comment by sonicxx** [comment]
There's also this:

https://github.com/sblakemore/cleanflight/commit/2d69dfd79df4f4fc4aa0aa9604e728f96c0231d2 Sorry if it's not reference properly, I'm a neophyte.

**Comment by jflyper** [comment]
@sonicxx Yeap. I've worked with him for a few weeks to integrate my bit-banged version with his stick control: cleanflight#1948

**Comment by jflyper** [comment]
So, we should at least find out which FX758-2 pin is connected to which M328P pins.
<img width="106" alt="fdaae298-1380-4c06-8973-881a54b3a9c0" src="https://cloud.githubusercontent.com/assets/14850998/19754752/492d0e38-9c4b-11e6-8c39-c151715bba7f.png">

**Comment by ShikOfTheRa** [comment]
The impulse source is here...

https://github.com/PickwickGroup/scarab-osd/tree/impulserc_vtx

**Comment by jflyper** [comment]
Ah, then it's MCU-less version of the SM21xT family.

**Comment by ShikOfTheRa** [comment]
Interested in adding - or shall I see if the original dev can add in support?

**Comment by jflyper** [comment]
Furious is said to upload and disclose their source for innova, which I believe (at least) functionally equivalent. Let's wait.

**Comment by ShikOfTheRa** [comment]
Actually I thought they were the same. My mistake. 
Innova != Impulserc Helix

But they might be functionally the same. 
Would be a nice addition.

**Comment by sniperxnl** [comment]
Can i do a firmware update on the novva form Furious with passtrough from fc?

**Comment by jflyper** [comment]
I don't have Innova, but in theory, yes.

However, updating Innova with current MWOSD will loose the VTX control capability.
This issue is for discussion toward the support of this capability.

**Comment by Rclf** [comment]
I was configuring the innova in mwosd and I accidentally pressed default . I can still see the osd in mwosd gui but the video is gone, it disabled the vtx part somehow

**Comment by jflyper** [comment]
@Rclf Did you try specifying NTSC/PAL in GUI?

**Comment by jflyper** [comment]
So, FuriousFPV is coming out with V2 of Innova. Any news on source code disclosure?

**Comment by Rclf** [comment]
Yes of course, this is not it.  My guess is that somehow it messed the firmware.

**Comment by jflyper** [comment]
If you've decided to stash it, can you do some investigation mentioned [above](https://github.com/ShikOfTheRa/scarab-osd/issues/327#issuecomment-256545317) ?

**Comment by ShikOfTheRa** [comment]
@rclf - suggest to ask your supplier to provide the source code - we can help you out then.

**Comment by proggod** [comment]
Has anyone attempted to contact them and request it?

**Comment by ShikOfTheRa** [comment]
Yes - I have and they said they would. I haven't received it yet though

**Comment by proggod** [comment]
How long has it been?

**Comment by ShikOfTheRa** [comment]
@proggod 2 months maybe. I have just emailed again.

**Comment by Rclf** [comment]
I am waiting for the firmware as well.  Should have it in a couple of days.

**Comment by ShikOfTheRa** [comment]
Had a mail from furious this morning - he is chasing the dev

**Comment by proggod** [comment]
Great, I asked someone to light a fire under them to get it to you :)

**Comment by TDIPWR** [comment]
Any word on the firmware?

**Comment by Rclf** [comment]
No not yet,  I emailed again yesterday  still no answer.

**Comment by DaveLister007** [comment]
I'm gonna push this little further - any news on source code from vendor?

If not, how different do you reckon this vtx is from this one??
https://github.com/PickwickGroup/scarab-osd/commit/9cb6e99da3446d763c82024485a4e6e6e0e7bd5e

Quick beeping with my multimeter revealed that CH1-CH3 pins of FX758 are connected via 10K resistors to MCU pins, so I'm guessing SPI. There are probably no dedicated pins for power settings, so it's probably set via SPI.

**Comment by jflyper** [comment]
@DaveLister007 
> If not, how different do you reckon this vtx is from this one??

Not much. You can probably come up with a working code only by substituting pins for spi to match that of innova.

**Comment by DaveLister007** [comment]
I messaged FuriousFPV and got the source code into email. There is nothing like sending something via serial port/SPI added, just raped PID menu for setting channel, and storing/loading it from EEPROM. Either im blind/dumb, setting channel/power is done via some other magic outsice MCU, or they deleted that piece of code before sending it to me. If anyone wants to have a look, let me know, i'll send it to you.

**Comment by ShikOfTheRa** [comment]
yeah. like to take a look. I'm hoping to add vtx support into mainstream code. If don't hear back from them its probably not going to be compatible.

**Comment by DaveLister007** [comment]
This is what I received. Seems to be based of 1.6 version. Rename it to .rar (githib doesn't allow that extension).
[FIRMWAVE.txt](https://github.com/ShikOfTheRa/scarab-osd/files/718014/FIRMWAVE.txt)

**Comment by u8mydog** [comment]
Try diffing the two OSD dirs. For example, globalVariables.h has EEPROM_ADR_BAND, EEPROM_ADR_CHANNEL, EEPROM_ADR_FREQ and EEPROM_ADR_POWER. Looks like serial.ino uses the eeprom values via PID settings.

**Comment by ShikOfTheRa** [comment]
Does appear that this does not include all the code..

Good to see the Piggy is stock to confirm what thought

**Comment by jflyper** [comment]
+1 This is a deteriorated code.

**Comment by DaveLister007** [comment]
So I messaged them again, and received some this. This one looks much better.
[MW_OSD_VTX.txt](https://github.com/ShikOfTheRa/scarab-osd/files/724622/MW_OSD_VTX.txt)

**Comment by ShikOfTheRa** [comment]
Well that took some time. It looks about right.

Can now consider which to implement. If any!

Not going to add anything new as closed doors for anything new for 1.7 . So maybe this is for 1.7.1

Thanks for the files. Hugely appreciated

**Comment by jflyper** [comment]
```
#define SPI_DATA_PIN    10
#define SPI_SELECT_PIN    9
#define SPI_CLOCK_PIN   8
```
This is what I wanted to know. There's nothing else special (except for hard coded EEPROM addresses).

And I've noticed that tx5823-pro code is used also.
[tx5823-pro](https://github.com/sheaivey/tx5823-pro/blob/master/src/tx5823-pro/tx5823-pro.ino)

**Comment by jflyper** [comment]
I'm now interested in how to control peripheral devices on OSD from FC...

Extend MSP_OSD message?

**Comment by DaveLister007** [comment]
@jflyper If the pins were only thing you were interested in, I could have told you month ago by reverse engineering it :)

Anyway, canvas mode and commands to set channels/power via FC is way to go IMHO.

**Comment by jflyper** [comment]
@DaveLister007 
It has been requested in the [6th comment](https://github.com/ShikOfTheRa/scarab-osd/issues/327#issuecomment-256545317)  above ;)

**Comment by HenryIvarsson** [comment]
Thank you guys for all the hard work! I flashed my BG Innova clone to the latest mwosd beta without thinking, should I try flashing back to the version posted above or wait for MWOSD support? When do you think there can be a version to test?

**Comment by jflyper** [comment]
@HenryIvarsson The current MWOSD doesn't support Innova's VTX capability. You should flash it back. The support schedule is not available yet (Probably after March 1st, when Betaflight 3.1.6 patch is released.)

**Comment by jflyper** [comment]
Okay, I can probably spend next couple of days for the initial cut of the Innova support.

**Comment by HenryIvarsson** [comment]
That sounds awesome, thank you!

**Comment by ShikOfTheRa** [comment]
@jflyPer - I have Vtx branch ready to merge for another  one. Can you wait until I merge? Might be a lot of common code can reuse. Need to review on my pic should be tomorrow or day after at latest

**Comment by jflyper** [comment]
Dah! I already did most of the code. Well, I'll pause as it is right now.

**Comment by HenryIvarsson** [comment]
Nice!! Is there any way to download a compiled version to test?

**Comment by HenryIvarsson** [comment]
Ah, sorry, missed the ShikOfTheRa's comment above, no stress guys! You guys rock!

**Comment by ShikOfTheRa** [comment]
@jflyper Dah! 2
Just synced a branch. Does this work for you - or possible to include its elements into your stuff. 

Used ImpulseRC code as base.


Menu separated from hardware type – to allow multiple hardware implementations

VTXHARDWARETYPE.ino will need: following and any associated code
  vtx_init() // set up hardware
  vtx_save()  // save vtx settings
  vtx_read() // load vtx settings from eeprom
  vtx_set_frequency(vtxBand, vtxChannel)  to change 

def.h
#ifdef VTX_RTC6705  | (add HW here) for some other hardware to enable menu

following settings are used:
  Settings[S_VTX_POWER]   = vtxPower;
  Settings[S_VTX_BAND]    = vtxBand;
  Settings[S_VTX_CHANNEL] = vtxChannel;

Its untested on hardware, so prob need a little debugging…

**Comment by macfly1202** [comment]
HI, I've got a genuine Innova vtx V1. It's perfectly working. But Essential issue I got is my cam is NTSC and innova - actual code - only support PAL so I can't see last line > impossible to navigate between menu in this case...I'm not alone in this case according to RCGroup thread. I'm sure there is solution on this problem like enumerate new page OR changing font in worst case (wendy3x5.h could do the trick, even if partial for some letters)  
So if needed 'tester' of alpha, beta version for Innova, I could made a try. I see the new VTX branch but doesn't know if could be test right now.

I've got a question : Does somebody have already try to "plug externally" micro or minimOSD via SPI wires like we do on this project https://github.com/markohoepken/rx5808_pro_osd/wiki/wireing-diagram. Pin are quiet easily accessible with soldering and don't understand why not using it instead of MSP ?

**Comment by jflyper** [comment]
@macfly1202 
The VTX branch is created without an actual hardware, so need some testing/debugging. And it's for ImpulseRC's Helix OSD, so must still be modified to tailer for Innova.

> I've got a question : Does somebody have already try to "plug externally" micro or minimOSD via SPI wires like we do on this project https://github.com/markohoepken/rx5808_pro_osd/wiki/wireing-diagram. (it's perfectly working on my HeadplayHD). Pin are quiet easily accessible with soldering and don't understand why not using it instead of MSP ?

I'm confused by this question.

- If I understand the link you've provided correctly, it's for receiver side configuration.
- In the wiring, SPI is controlling MAX7456 and RX5808, controlling devices.
- MSP (and other serial protocols) are used to communicate with data source (e.g. FC), to receive what to display as overlay.

And what do you mean by "Using SPI instead of MSP"?

**Comment by macfly1202** [comment]
@jflyper  Thanks for clarification about VTX branch.
About 
I'm sorry If not very clear. I'm not developper just DIYer. If I well understand, Omnibus F4 pro or betaflight F3 board display infos receive (and transmit setup) with SPI connection instead of traditionnal "serial" uart connection. In my example link, SPI from 328p is use to display and change channel of rx5808. But I was simply asking instead of wiring micro/minimOSD with uart connection why not replacing UART with  SPI wiring from FC to 7456 with multiples advantages (freing 1 uart & DMA on >=F3 & cpu).
Perhaps it's a nonsense for you but simply asking. Thank you very much for all the great work you done.

**Comment by jflyper** [comment]
FC-integrated MAX7456 boards use SPI to control the MAX7456.

FC-integrated MimimOSD-variant use UART to communicate with M328P, M328P use SPI to control the MAX7456.

So, what are you proposing?

FC-integrated MinimOSD-variant to use SPI to communicate with M328P?

**Comment by macfly1202** [comment]
NOT integrated FC use SPI to control the MAX7456 in 1st time. Why must it be integrated ? 3-4 wires to solder between FC F PIN and MAX PIN.   
In 2nd time, FC use SPI to control RX.

**Comment by u8mydog** [comment]
I have wondered the same thing if I understand the original question.
Give minimOSD the option to enable FC integration by using SPI communication from FC -> M328P -> MAX7456. 

Basically, the M328P acts as a simple bridge to the MAX7456 moving all OSD functionality to the FC instead of the M328P.

**Comment by HenryIvarsson** [comment]
Hey guys, any code ready that I should test on my Innova yet? If so I might need som help compiling if it is not posted somewhere already.

**Comment by jflyper** [comment]
@HenryIvarsson [Read this](https://github.com/ShikOfTheRa/scarab-osd/issues/327#issuecomment-280391069)

**Comment by jflyper** [comment]
@u8mydog 
> Basically, the M328P acts as a simple bridge to the MAX7456 moving all OSD functionality to the FC instead of the M328P.

If we do this via SPI, we don't need M328 at all.
If we do this via UART, it is too slow, at least atm. We have tested this option during the development of the CMS over MSP_DISPLAYPORT; too much overhead sending characters instead of sending binary data.

**Comment by HenryIvarsson** [comment]
@jflyper Thank you, just so I understand correctly, I assume I cannot test Innova on the Impulse settings on the vtx branch as the hardware is to far apart?

**Comment by ShikOfTheRa** [comment]
@HenryIvarsson - nothing available for Innova. Can't use current branch
The Furious guys added that in. I am planning to add it in, but have not started yet... 
also do not have any hardware to test it with.

**Comment by HenryIvarsson** [comment]
Ok, then I understand, thank you

**Comment by ShikOfTheRa** [comment]
@jflyper how would you like to progress the work you already done with the vtx branch? I'm happy if you would like to scrap what I have put in and put yours instead?

**Comment by macfly1202** [comment]
@ShikOfTheRa : I made a little paypal contribution that could help in hardware issue side: "do not have any hardware to test it"

**Comment by DaveLister007** [comment]
Will the current implementation support MSP_VTX command? (Or whatever is the command, that betaflights sends via MSP to VTX to set channel called)?

**Comment by jflyper** [comment]
@DaveLister007 Something like that planned, but not implemented yet. The MSP actually wasn't designed for this kind of operation...

**Comment by ShikOfTheRa** [comment]
@macfly1202  - that is awesome of you. I'll get on to furious. I think their HQ is close to me although different country and a few KM of water between us.

**Comment by ShikOfTheRa** [comment]
@jflyper - I didn't know about the betaflight MSP VTX support. If appropriate add it in.
Not worried about menu ordering. Feel free to make it flexible. 
I have a helix but waiting on rest of gear on slow boat from china to build a test setup.

**Comment by jflyper** [comment]
@ShikOfTheRa MSP VTX stuff is still a vaporware.

BTW, did you stash simple DEBUG and FORCEDEBUG away in favor of DEBUGMW?

**Comment by macfly1202** [comment]
(@ShikOfTheRa :fyi there a new version of innova (v2) with different layout pcb but seems to have same features. Don't know if Banggood clone is v1 or v2 but must be v1. Most of seller have v1 like picture on https://www.unmannedtechshop.co.uk/innova-osd-vtx-board/   (v1 : oscillator is parallele to pcb side and v2 oscillator is @45 degree of pcb side))

**Comment by HenryIvarsson** [comment]
@macfly1202 good idea, I also pitched in. @ShikOfTheRa I have the BG clone, it's v1 for sure. V2 takes lipo voltage but is sold out until end of march on Furious site, that's why I got the clone in the first place. I might now where to find them in stock, just confirming they are v 2 first. I will sound off when I know.

**Comment by jflyper** [comment]
Does anybody have a user manual for Helix VTX, or pointer to it?

I suspect it handles stick commands that is compatible with SINGULARITY...

**Comment by jflyper** [comment]
Okay, Helix should be working with #382 (PR to VTX branch).

Call it a day 😫  Bed time!!! 😴

**Comment by jflyper** [comment]
@all Do Innova has any LED onboard, except for power indicator LED?

**Comment by ShikOfTheRa** [comment]
there are 2 onboard not sure if one is power.

**Comment by ShikOfTheRa** [comment]
Starting build of mine tomorrow. Not got some of the kit yet. But should be enough to test

**Comment by macfly1202** [comment]
On geniune furious v1 there is only 1 led (green) always on when powered. (Linked to 3.3 ? voltage reg as it seems) 
![img_6824](https://cloud.githubusercontent.com/assets/6387630/23092089/ce047ebc-f5c3-11e6-8639-52f7b80ae77c.JPG)

**Comment by jflyper** [comment]
@macfly1202 Thanks!

**Comment by jflyper** [comment]
Innova support is ready for alpha testing.

https://github.com/jflyper/scarab-osd/tree/vtx-mods-for-innova

**Comment by macfly1202** [comment]
@jflyper I will in the afternoon - need to play with my little baby first. Thanks a lot.

**Comment by macfly1202** [comment]
@jflyper : is Vtx power still adjustable or not ? I ever find vtx is a boiler. I will test power output with my RF explorer spectrum analyser if I find an ufl to sma on my shelve...

**Comment by jflyper** [comment]
@macfly1202 Well, actually, the power adjustment wasn't tested (as stated in PR) 😩 

I'll look into it.

**Comment by macfly1202** [comment]
@jflyper : like I said previoulsy, I have got the hardware necessary to test it (RF Explorer 6G ) so I will test the power and frequency allocation just in case and make feedback before end of Weekend.

**Comment by jflyper** [comment]
@macfly1202 Thanks, I really appreciate it 👍

**Comment by macfly1202** [comment]
@jflyper  : I try to complile it under arduino 1.0.6, 167,  169. Compilation error.
`In file included from MW_OSD.ino:90:
Def.h:545:1: warning: "MENU_VTX" redefined
Def.h:62:1: warning: this is the location of the previous definition
In file included from RTC6705.ino:3:
wireMacro.h:2:1: warning: "ATOMIC_BLOCK" redefined
In file included from MW_OSD.ino:88:
/private/var/folders/4p/t65kmy8j0dz98jw2_md8l1080000gn/T/AppTranslocation/E3F694D5-974F-45E2-B749-BF0A518305E4/d/Arduino_106.app/Contents/Resources/Java/hardware/tools/avr/bin/../lib/gcc/avr/4.3.2/../../../../avr/include/util/atomic.h:203:1: warning: this is the location of the previous definition
In file included from RTC6705.ino:3:
wireMacro.h:27:1: warning: "_BV" redefined
In file included from /private/var/folders/4p/t65kmy8j0dz98jw2_md8l1080000gn/T/AppTranslocation/E3F694D5-974F-45E2-B749-BF0A518305E4/d/Arduino_106.app/Contents/Resources/Java/hardware/tools/avr/bin/../lib/gcc/avr/4.3.2/../../../../avr/include/avr/io.h:99,
                 from /private/var/folders/4p/t65kmy8j0dz98jw2_md8l1080000gn/T/AppTranslocation/E3F694D5-974F-45E2-B749-BF0A518305E4/d/Arduino_106.app/Contents/Resources/Java/hardware/tools/avr/bin/../lib/gcc/avr/4.3.2/../../../../avr/include/avr/pgmspace.h:82,
                 from /private/var/folders/4p/t65kmy8j0dz98jw2_md8l1080000gn/T/AppTranslocation/E3F694D5-974F-45E2-B749-BF0A518305E4/d/Arduino_106.app/Contents/Resources/Java/hardware/arduino/cores/arduino/Arduino.h:27,
                 from MW_OSD.ino:32:
/private/var/folders/4p/t65kmy8j0dz98jw2_md8l1080000gn/T/AppTranslocation/E3F694D5-974F-45E2-B749-BF0A518305E4/d/Arduino_106.app/Contents/Resources/Java/hardware/tools/avr/bin/../lib/gcc/avr/4.3.2/../../../../avr/include/avr/sfr_defs.h:208:1: warning: this is the location of the previous definition
RTC6705.ino: In function 'void vtx_transfer(int, int, uint32_t)':
RTC6705:104: error: '_delay_us' was not declared in this scope
RTC6705:110: error: '_delay_us' was not declared in this scope
RTC6705.ino: In function 'void vtx_set_frequency(uint8_t, uint8_t)':
RTC6705:173: error: '_delay_us' was not declared in this scope
RTC6705:179: error: '_delay_us' was not declared in this scope
Screen.ino: In function 'void formatDistance(int32_t, uint8_t, uint8_t)':
Screen.ino:2269: warning: unused variable 'tmp'
Serial.ino: In function 'void serialMSPCheck()':
Serial.ino:359: warning: unused variable 'canvasa'`

**Comment by jflyper** [comment]
@macfly1202 Can you try with 1.6.12?

**Comment by macfly1202** [comment]
@jflyper it s what I'm just doing now ! It seems to be ok. I will post feedback soon.

**Comment by Rclf** [comment]
Does anyone know if it's possible to burn the bootloader to the innova ?  I keep getting failed to talk to the bootloader it is like it's bricked but I still can see the osd in mwosd configurator, but no video.

**Comment by macfly1202** [comment]
@jflyper :arduino 1612 say "upgrade" finished (but take age in comparaison of master branch). At innova vtx startup still write "MW-OSD - R1.6". according to source code it must be "MW-OSD - R1.6.8" if I understand well. I think something fail and innova isn't flashed as I initially think. I will try under PC later (beurk) or linux. This test was on mac.

**Comment by HenryIvarsson** [comment]
Trying to test on my Innova; on my mac, using Betaflight passthru as my Innova clone is soldered in deep, got a promising "Done Uploading" from Arduino 1.6.12. Did no changes in config.h as Innova was defined as default. Uploaded digital.mcm in the Chrome app from @jpflyers Innova branch. My old config from the latest regular mwosd was still there so no other changes in the web app. Booted the Hedwig, osd reports MWOSD 1.6.8 Tried to find the vtx channel and power config in both the yaw left and yaw right osd menu but unable to find it. What am I doing wrong?

**Comment by jflyper** [comment]
@HenryIvarsson It should be between "ADVANCED TUNE" and "STATISTICS".

**Comment by jflyper** [comment]
@Rclf From pics of Innova on the net (which are mostly of the non-RF module side), I can't find any ICSP pads. It must be on the RF module side. Can you find any unoccupied pads?

**Comment by jflyper** [comment]
@all The up to date branch is at
https://github.com/jflyper/scarab-osd/tree/vtx-fix-bad-power-writes

**Comment by macfly1202** [comment]
@jflyper:  I tested all standard process with dtr solder blob and genuine ftdi or cp2102. I even made some short to reset pin directly. Something is bad in design of genuine innova or my board has some problem. I want to add I'm not a noob on flashing firmware on atmel and stm. 
Finally, bored to try all standard I decided to use an AVR on 328p directly. I compiled with arduino 1.6.12 and get the hex. With avrdudess, I sucessfully flash the firmware and mwosd 1.6.8 was displayed at starting of pikoblx/innova. Here screenshot of wiring (little tiny soldering session as I love them). It seems all features are ok and working. I didn't find the time one by one all of them but seems ok. Only problem is with vtx 758 power seems to be at 25mw despite 200mw selection. Band and channel seems to be ok according to my rx5808 pro mod on headplay. You made an excellent work : congrats ! I will post DVR of test session later. I need to go for business trip now. 
![img_6977](https://cloud.githubusercontent.com/assets/6387630/23104137/707c71fe-f6c8-11e6-86c7-b516fabce340.JPG)
![img_6978](https://cloud.githubusercontent.com/assets/6387630/23104139/70a4839c-f6c8-11e6-89ef-878a69946ebd.JPG)
![img_6979](https://cloud.githubusercontent.com/assets/6387630/23104138/70a4a4bc-f6c8-11e6-8d5b-7e7edf1a8074.JPG)
![img_6980](https://cloud.githubusercontent.com/assets/6387630/23104140/70a5e156-f6c8-11e6-8430-38754d8c2364.JPG)

**Comment by jflyper** [comment]
@macfly1202 Wow! So you actually broken out the ICSP signals!

And thanks for testing!

**Comment by macfly1202** [comment]
@jflyper : what do you mean by "broken out ISCP signal line"? I'm not understanding. Arduino said burning was ok with ftdi but it was not the case. Another user reported here some similar situation. I'm use to meet tricky bootloader and always find solution. I'm nearly sure the banggood clone is ok with bootloader contrary to genuine one.

I test the initial commit. See after there was a new release. I will be away for few days so impossible to test new one till friday. Sorry about that and thanks again.

**Comment by jflyper** [comment]
@macfly1202 Ah, "to break out" is to attach a lead or trace to a chip, and I was kinda surprised that Innova does not have ICSP pads (as predicted in my previous [comment](https://github.com/ShikOfTheRa/scarab-osd/issues/327#issuecomment-280893842)).

And what I wanted to say is "Great micro soldering work!" :)

**Comment by macfly1202** [comment]
Ahh ok ! I'm a DIYer lol. Here more info could help some people. For brave people who want to burn with usbasp: it's very difficult soldering and must have good iron, not overheating (max max  300degree for 5-6 sec):
Mosi is physical pin15 on atmel 328p, 2nd horizontal left from bottom right linked to max7456  6th "pin" (miso) from bottom side left. (Solder/connect usbasp mosi here)
Miso is physical pin16 on 328p, 1st horizontal left from bottom right linked to max7456 4th "pin" (mosi) from bottom side left. (Solder/connect usbasp miso here)
Sck is physical pin17 on atmel 328p, 1st vertical up from bottom right linked to max7456 5th "pin" (sck) from bottom side left. Personnally, it's nearly impossible to solder pin 5 between 4 and 6 on max 7456 (I difn't use official pin of max7456 in my explain because haven't them now). So I prefer to use sck from atmel and because pin are shorter on 328p than max7456, I always prefer to use miso and mosi on max7456. 
(I haven't got the atmel mega socket burner from hobbyking, the white one for esc, and not sure it s ok for this package side, not sure there is enough space for it with the 2 470 uFcondensator and 20$ is quiet expensive I find : https://hobbyking.com/en_us/atmel-atmega-socket-firmware-flashing-tool.html)

**Comment by jflyper** [comment]
@macfly1202 Did you burn an arduino bootloader, or sketch hex?

**Comment by macfly1202** [comment]
@jflyper : I simply burn the sketch hex found in compile folder, display in arduino with verbose mode (there is 2 hex one with bootloader in his name, the other I choose without bootloader in his name)

**Comment by macfly1202** [comment]
video from dvr test of Innova firmware alpha release : https://youtu.be/18YE5cQseE0 with a crappy youtube song like we love so much (lol!) instead of hmdvr chaos noise.

**Comment by HenryIvarsson** [comment]
Nice video! I got it to flash, had to solder in leads to external serial adapter to get dtr and cut 5v pin back feeding voltage to piko as a booted piko made chaos with my flashing. Got the vtx menu, very nice. However, changing to E4, typically puts me on 5645, similar to my 9 other quads. Setting to e4 here took me to 5805ish and one hour spent trying other band and channel combos couldn't even get me close to 5645?

**Comment by jflyper** [comment]
@macfly1202 Nice vid! You tested the transition into BF CMS, too! :)

@HenryIvarsson Let me take a look at the frequency table.

**Comment by jflyper** [comment]
I want some review on this write up.

Innova flashing tips
(1) Disconnect from FC or power it down while flashing.
FC's serial signal interferes with external serial adapter used for flashing.

(2) Make sure DTR is correctly connected.
Innova V1: Break out the DTR pad.
Innova V2: DTR-LEDstrip solder jumper to open. De-install if jumped.

- DTR is used to put the Innova into bootloader mode.
- On FTDI USB-Serial adapter, corresponding signal is RTS# (Pin 6, GREEN).

**Comment by HenryIvarsson** [comment]
Thank you @jpflyer will try the latest tonight. Write up looks good, spelling out that flashing thru betafligt will not work due to missing dtr connection might make sense.

**Comment by macfly1202** [comment]
@jflyper: I tested the 2 process for flashing. 
1/ thru betaflight with solder blob on dtr pin
2/ with cp2102 and genuine FTDI 5v, 3.3v on TTL lines,  with DTR/RST plugged line with solder blob
3/ same direct wire between DTR wire and RST physical pin of atmel (the red wire on my photos) 
--> no method above has been sucessful. Fitmware is flahed with arduino (anormally long time) but it seems eeprom param aren't reset and bootloader fail. Perhaps after burning bootlooader through usbasp, this method above will work.

**Comment by jflyper** [comment]
@macfly1202 People's experience suggest that there are apparently ones with bootloader and ones without?

**Comment by macfly1202** [comment]
@jflyper : I will test to burn bootloader Friday - I think you are true: it must be the explanation of not successful burning - can't test that before because away fom home. 
Does people with innova clone from Banggood have succesfully change firmware ? (I read somewhere its was the case - sometimes clones are better the genuine...lol)

**Comment by HenryIvarsson** [comment]
@macfly1202 I could flash my BG clone with cp2102 serial adapter, remains to be seen what my channel map looks like. I do have a blob on the dtr pad enabling flashing, do you know if I should be able to flash thru BetaFlight also? If so I might wanna try that again. That would be super convenient.

**Comment by macfly1202** [comment]
@HenryIvarsson : si it's confirm what I already read somewhere > Innova clone from BG have bootloader in contrary to some genuine Innova from furious like mine. 
For your question with serialpassthrough, honestly I don't know but DTR of innova must be mapped to DTR of piko or other FC. Don't know if DTR used with VCP driver. I don't know which PIN LedStrip is mapped and if used with serialpassthrough.
So in conclusion : doesn't know ! ask in furious rcgroup forum ?

**Comment by jflyper** [comment]
If the VCP driver on FC side can be modded to handle DTR signal transitions, it can be translated to GPIO high/low by adding a new feature, may be by resource mapping, e.g.: "resource dtrpassthrough a08".

**Comment by jflyper** [comment]
Updated the above writeup.

BTW, I found a couple of suspicious artifacts in the pic.
![b1f4f0fd-d07d-4d97-a249-383c065d1fcb](https://cloud.githubusercontent.com/assets/14850998/23132362/b1b26774-f7d0-11e6-9e58-f34aa2cd5ac0.jpg)

**Comment by macfly1202** [comment]
@jflyper: I didn't have the board with me but think it's hot glue blob.

**Comment by jflyper** [comment]
@macfly1202 Ack!

**Comment by macfly1202** [comment]
@jflyper : just want to correct you excellent Tuto : the green wire is RST/DTR and not ground. The tiny red wire is a bypass off resistor and condensator to directly connect to RST physical pin of atmel 328p. 
![img_7002](https://cloud.githubusercontent.com/assets/6387630/23140427/de0548e2-f7b1-11e6-92eb-53b24135911c.PNG)

**Comment by jflyper** [comment]
@macfly1202 Oh! Is bypass wire required?

**Comment by macfly1202** [comment]
@jflyper : not sure but always prefer direct connect with RST for ftdi/usbasp. I made this before doing it with usbasp.

**Comment by jflyper** [comment]
@macfly1202 Pic in the how-to has been corrected.

I believe that the bypass is required for ASP case in which direct access to MCU RESET is required.

**Comment by macfly1202** [comment]
@jflyper: thanks it was what I believe. I will test bootloader burn tonight to see if after that ftdi firmware upgrade work.

**Comment by macfly1202** [comment]
@jflyper : so I'm ready to make more test. Where is the last up-to-date branch ? here or on you github ?

**Comment by jflyper** [comment]
@macfly1202 It's all on the VTX branch now! :)

**Comment by macfly1202** [comment]
@jflyper I've some preliminary test with VTX branch. I simply uncomment FFPV_INNOVA in Config.h. Don't know if there is anothers param to change for Innova.  
1/ about burning bootloader : I didn't succesfully achieve it for now. Will do some attempt this WE
2/ about firmware : there is in issue in power mode. Innova is always in 200mW (223mW on 5760) despite validating in OSD the 25mW. I made many attemps with rebooting without having the correct 14 dBM (25mW).
Thanks again for this great work ! love it ! The issue of cropping osd is now solved with another page menu dedicated for vtx. Is there a way to display band and/or frequency and/or power on OSD  screen, like in setup menu ?

**Comment by Rclf** [comment]
it seems the innova is discontinued , no longer on the site.

**Comment by jflyper** [comment]
@macfly1202 I will look into the power control code again. And yes, we can have band/chan/freq/power displayed on the opening screen; I will add that as well once I get back to my bench (which is 8,000km away now).

@Rclf WTH!? Hmm... it's gone :thinking:

**Comment by macfly1202** [comment]
@Rclf : It could be a bug on their website update because Innova v2 has been on pre-order. It will be strange to cancel order. Innova v1 is stopped for sure. There is still the BG clone in all the case.   
@jflyper : thanks ! nice travel...

**Comment by Rclf** [comment]
@macfly1202 I just received a v2 two hours ago as a replacement for the faulty v1, its strange indeed.

**Comment by jflyper** [comment]
@macfly1202 If you have time, can you try this branch?
https://github.com/jflyper/scarab-osd/tree/vtx-mod-20170224

It has DEBUG defined, and should display 4 debug variables, band, channel and power from the top (last one is unused).

And if you disable the DEBUG (Def.h), you might see the information about vtx on the opening screen.

**Comment by macfly1202** [comment]
@jflyper : I made a quick short test and unfortunately, output power doesn't seem to change. Here what I saw with new firmware upgrade of this day: https://youtu.be/rlelOl6lG7U

**Comment by macfly1202** [comment]
@jflyper: I test this last version and power seems to be ok now. Great thanks=> Power of vtx is changing from 25mW to 200mW. I made a custom uFl to SMA and it's probably why my measure on 5740 or 5865 oscilating a lot. I need to make another test with floor noise but I found far more than 200mW (320mW but not stable due to overheating of the unit/ Low power was approx from 5mW to 30mW for same reason). I  would like to add I'm not sure to my previous measure made without direct uFL-SMA connection (was throught "air" with 2 circular polarized Ibcrazy). Essential is power is well changed and I would like to thank you very much for this job. I've got a dvr video of debug value I will post tomorrow. Important notice : power of vtx is changing a lot according to camera that is connected on innova. change between camera not plug TO plug : resultat -3dbm (with camera) and shifting frequency +0.4MHZ approx (with camera) 5800Mhz shift to 5801.3 instaed of 5800.9Mhz

**Comment by jflyper** [comment]
@macfly1202 
> I test this last version and power seems to be ok now. Great thanks=> Power of vtx is changing from 25mW to 200mW. I made a custom uFl to SMA and it's probably why my measure on 5740 or 5865 oscilating a lot.
Okay, so the logic is fine.

> I need to make another test with floor noise but I found far more than 200mW (320mW but not stable due to overheating of the unit/ Low power was approx from 5mW to 30mW for same reason).

There are magic numbers for what's so called "_Pre-driver and PA Control Register_". The current value is taken directly from FFPV's original source code. However, there are other values in other implementations such as SIRINFPV, which claims same 200mW output (probably) using the same FX758-2 unit. I will probably mod the current source for the investigation of what these values actually do.

>   Important notice : power of vtx is changing a lot according to camera that is connected on innova. change between camera not plug TO plug : resultat -3dbm (with camera) and shifting frequency +0.4MHZ approx (with camera) 5800Mhz shift to 5801.3 instaed of 5800.9Mhz

I suspect a change in supply voltage to the internal RF chip (RTC6705) that is causing power down (PA) and freq shift (VCO).

**Comment by OUFanInCo** [comment]
> Rclf commented 6 days ago
> it seems the innova is discontinued , no longer on the site.
-----------------
Apparently, it has been discontinued. I can order one from other sources, but I need to know the V2 design corrected most of their issues? They won't say.

Sales at FuriousFPV agrees with this:

---------------------------
>Hello,
>I was on your site tonight, ready to order an Innova V2 for a new build and I see this:
>Catalog » FPV-0064-S
>Product not found!
>Please don't tell me that it's been discontinued already. I was really looking forward to making this part of my new build.
>Thanks
----------------------------------------------------------------------------------------
>Sales FuriousFPV
>Feb 26 (3 days ago)
>	
>to me
>Hi,
>Sorry for the inconvenience but yes we have discontinued the Innova.
>Best Regards,
>Lena
------------------------------------------------
I requested more clarification over Facebook and I've only heard that they will be releasing a product to replace it "soon". 

----------------------------------------
>Sun 12:05pm
>   **FuriousFPV:** No, we just release new version adjustable power from 25-200mw with pid mode and smaller form factor so we do not want to make old versions anymore.
>Sun 1:32pm
>   **Me:** I know you've discontinued the Innova v1.0. I thought I was looking at the Innova v2.0, which is exactly as you describe. Adjustable from 25-200mW. I can't find another 20mm VTX-OSD on your site. Could you provide me with a link?
>   **FuriousFPV:** It's been discontinued apparently.
>   **Me:** Really? Because of issues? I guess I should skip out on ordering it from HK also.
>   **FuriousFPV:** We will have another one soon
>   **Me:** Ok. I'll keep an eye out for it.
--------------------------------------------------

**Comment by OUFanInCo** [comment]
PS - Furious FPV might be tied up in the following issue and not have time to respond to other things at this point:
[https://www.rcgroups.com/forums/showthread.php?2839452-Furious-True-D-V3-Diversity-Receiver-System-Kills-Fat-Shark-HD-Dominators](url)

**Comment by ShikOfTheRa** [comment]
Oh that's crap scenario. I guess similar products will have same issue.

Got to feel for Fatshark - Guys loading them up their products higher that original design intent. Also to Furious - great product. Probably did loads of testing an no issues... assumed fatsharks would be able to source power. 
Sort of thing you wouldn't really think about until someone figured it out later down the road.
Fatshark are well within their moral obligation to charge for repairs IMHO. I bet loads put original module back in and send them in like that... and unless the guy doing it is honest and up front they will have to do any warranty repairs...

**Comment by ShikOfTheRa** [comment]
Kudos to Furious in how they are handling this... I just saw their response....

**Comment by slopeinsb** [comment]
After hitting 'default' in MWOSD which appears to disable VTX - and then attempting https://www.youtube.com/watch?v=U_-RPSC91CU I got a "failed to talk to bootloader" error. 

Is there any hope without having to do the micro soldering in order to flash these?

**Comment by ShikOfTheRa** [comment]
That in the video isn't mwosd...

You should talk to furiousFPV for any support of Innova as its a custom build they have created. There is currently no official MWOSD support for it

**Comment by slopeinsb** [comment]
Understood - I used MWOSD without knowing the board wasn't supported - it 'appeared' to be - and then while tracking down the issue I ended up here, and saw others had done the same.

**Comment by kelleysislander** [comment]
Hi All,

I found you all while looking for a version of MW_OSD that would restore the VTx menu items.  I was searching n"Innova OSD firmware upgrade" 

Anyways, the board works correctly in every other way, Is there a branch  or version I could flash to allow the VTx control settings to show up in the OSD menu?

Thanks,   Bill

**Comment by yds** [comment]
@ShikOfTheRa @jflyper any chance the VTX branch will get merged?

I've installed the [VTX branch](https://GitHub.com/ShikOfTheRa/scarab-osd/tree/VTX) from the latest sources configured for [dRonin](https://GitHub.com/d-ronin/dRonin) and all seems to work as expected on a couple of the [Banggood clones](https://www.banggood.com/Micro-20x20mm-VTX-OSD-Board-40CH-5_8G-25mw-200mw-Mini-FPV-Transmitter-Integrated-OSD-RHCP-p-1107269.html) that I have.

Thanks much for making it work! This hardware wouldn't be worth much without all of your work.

**Comment by jflyper** [comment]
@yds 

> any chance the VTX branch will get merged?

Yes, of course. Thanks for reminding me!!!

**Comment by erwieser** [comment]
Hello everyone,

I too hit 'default' without thinking or reading this before and since then my VTX isn't transmitting anything at all.
So I tried to flash (with CP2102) the latest VTX branch to my Innova V1 clone (not from BG but I guess it's the same clone) with the hope that I'd see something again. The sketch is uploaded successfully (with Arduino 1.6.5) but there is still no picture at all :(
Communication with Betaflight 3.1.1 works, MWOSD GUI works and I can upload a font but there is no OSD or picture appearing at all (currently I have no camera but the OSD should be transmitted anyways right?)

What am I doing wrong? @HenryIvarsson @yds  How did you manage to get it working?
Did I connect the DTR pin of the CP2102 correctly?
![flash_innova](https://user-images.githubusercontent.com/29452152/27192725-a7054f2c-51fc-11e7-9bf0-e6493dffe81c.jpg)

Thank you guys very much in advance! Already appreciate your hard work!

**Comment by yds** [comment]
@erwieser shorting the DTR pads with a blob of solder turns the 5th (LED) pin into a DTR pin -- there is no need for the wire you connected to the DTR pads.

FYI, that 5th (LED) pin there is just a pass through to the 3 pin led connector.  You can remove the solder blob shorting the DTR pads to revert back to LED pass through functionality.

I've been told that there's a short window right after power up when these sorta boards can be flashed without needing the DTR signal. Anyone get that to work?

Note how the TX/RX (Green/Blue) wires are crossed -- the INNOVA manual labels them backwards (or maybe they meant to label the pins to connect to on the USB adapter?) in any case that's the harness I use to flash and configure the VTX-OSD:
![osd-vtx-ftdi](https://user-images.githubusercontent.com/86074/27203976-727a0992-51f6-11e7-8bb8-762b5204a663.jpg)

**Comment by erwieser** [comment]
My wiring looks (almost) the same. As I have no DTR pin on my CP2102 module I had to do some micro soldering:
![20170616_110553](https://user-images.githubusercontent.com/29452152/27222575-13ba885e-528d-11e7-9d1b-5e6797b8f56b.jpg)
In the last picture I soldered the red cable (DTR from CP2102) directly to the solder blob, now it's connected with the harness. So it's basically the same, right?

Tried to do it without the DTR pin but that just results in an error. Where did you hear/read that?
I also tried the method described [here](https://www.youtube.com/watch?v=U_-RPSC91CU) (thanks @slopeinsb) and it didn't result in an error but there is still no picture showing up at all.

Slowly I start to ask myself whether the VTX has a hardware issue or not. After flashing, when I power it with a battery and try to see something the VTX becomes pretty hot. I read that this is normal if it's transmitting with the default 200 mW, but then why can't I see anything?
Is there something I can check or set manually on the VTX or the Atmega 328 over the serial port maybe?

**Comment by macfly1202** [comment]
Honestly @erwieser, why spend so more time with FTDI that didn't work for so many people. I've spend many hours on trying flashing with 3 different ftdi. The only - and very usefull tool to use and add on his shelve - is an usbasp. It will cost you 5-15$ (always take with 3.3 and 5v)  
Check my post above. Don't use the DTR pad blob, on my piko version 1  there was a loop back to atmel that gave me more issue it's why I use another way to go to reset physical pin 29 of atmel328.
so buy an usbasp, flash it like I do, and go flying....or buy a banggood clone that didn't seem to have this issue...when clone is better than original...

**Comment by erwieser** [comment]
@macfly1202 ESCs and camera for this build are still on their way so unfortunately I can't just go and fly yet.
But since I messed this one up, I'm really eager to fix it again somehow.

Since I only have the CP2102 by now: @HenryIvarsson @yds How did you manage to flash your clones? Which version of Arduino did you use? What did you uncomment in config.h? What was your whole procedure to success?
Otherwise, the 3,60€ USBasp is on its way and I'll see how I get along with that next week...

**Comment by raemin** [comment]
Just tested the latest VTX branch:
- the default power value should definitively be set to 25mw as the VTX is likely to fry when sitting on the bench without airflow.
- got some warnings during compile : **sketch\Def.h:991:0: warning: "INFO_CONTROLLER" redefined  #define INFO_CONTROLLER 5 sketch\Def.h:988:0: note: this is the location of the previous definition   #define INFO_CONTROLLER 4** It seems there is a confusion between Taulabs and dRonin (no error when selecting taulabs instead of dRonin)
- the provided MWOSD configurator does not have a VTX menu.

**Comment by ShikOfTheRa** [comment]
I have attempted to merge this into the master.

**Comment by hobbya** [comment]
Tested 1.7.2.2 successfully on the bench but no flight test yet .

Hardware configuration: Common minimsod board hard wired to Common RTC6705 module SPI port
MWOSD configuration: FFPV_INNOVA + APM + BAUDRATE 57600
Flight Controller: Pixhawk
RF equipment: RF Explorer spectrum analyzer

Flight modes, compass, satellites, voltage, current, RSSI: ok
Channel switching: all 40 channels ok (Boscam A, B, E, Fatshark, Race)
Power switiching: 25 or 200mW: ok

Some other observations:
- Arduino alerted low memory availability which could result in stability issues
- If the 3 wires between minimosd and RTC6705 got loose or disconnected, the osd screen flickered and finally blackened until I disconnected the telemetry port of flight controller.

**Comment by Arise** [comment]
How do I flash this? I think I have an original version of Innova VTX, v1.

First asking for the software part, I assume I need to compile from source?
What do I need to overwrite and from where?

Thanks, I'm a noob, I've read the whole post, I understand I need some soldering, but can't understand how to build the sketch (with what files) in arduino.

**Comment by unitware** [comment]
Hi, I have this board, is that supporter to work? I have not been able to connect with mwosd-gui nor program it from arduino. Any help is appreciated, i’d like to get it to work with inav. 

![40E710FA-861C-4DC1-84A7-EA56D6483882](https://user-images.githubusercontent.com/6065378/86622539-517d1780-bfc0-11ea-9e6c-dc6658f862d5.jpeg)
![9CFDCE6E-B2B1-43A4-8440-3E5C5C12279E](https://user-images.githubusercontent.com/6065378/86622545-5641cb80-bfc0-11ea-9ae4-b4c09b4a2015.jpeg)
![04DCB260-B75C-4E0F-ACB8-B4A8306521D5](https://user-images.githubusercontent.com/6065378/86622547-56da6200-bfc0-11ea-9293-248349609c73.jpeg)

**Comment by error414** [comment]
Have you tried swap TX/RX wires?

**Comment by unitware** [comment]
@error414 yes I have tried using a 300 ohm resistor on tx to not burn the uart pins

**Comment by ShikOfTheRa** [comment]
First thing - need to get it working from Arduino - but maybe ask FFPV for help. Below are the most common issues. It really is 99% ....

1 Unable to flash and I get "Programming FAILED: sending 3020"

99% of time this is connectivity or power issue
Check you are selecting the correct comm port
TX of OSD must connect to RX of programmer
RX of OSD must connect to TX of programmer
Programmer requires all six pins connected (Well, at least 5!)
Nothing else must power the OSD when programming. Only the USB programmer

Occasionally programmers / OSD have tx/rx mislabelled...

**Comment by unitware** [comment]
Thanks, this is probably one of those without bootloader. Now I'm struggling to connect an arduino-isp and program it before the wires come loose. My eyes are not up to par with this and I do not have a microscope. Patience is hard

**Comment by unitware** [comment]
So, finally some progress. I managed to flash the board via arduino isp, it was hard to get the tiny soldering right but i could get a bootloader and then the application in there. When I had some life in the board it turns out that the serial tx from the board was broken. Thus my attempts to get the arduino bootloader was futile.

But I have two boards and with the other the bootloader worked from scratch, excellent!

However: 
- The page with what I presume vtx settings is corrupt (see picture). I have tried the LARGE font, and the DEFAULT one.
- The Main voltage from FC is about half of the value that it should be, the voltage is correct when looking at the inav configurator. 

Any feedback is appreciated :)

![IMG_4311](https://user-images.githubusercontent.com/6065378/86829466-ba35c400-c094-11ea-8f85-3d3c1a606ed5.JPG)

**Comment by raemin** [comment]
I've been using this board (a V1 clone actually) for years on MWOSD v1.7.2.0, works like a charm.  All the issues related to the "stock" FuriousFPV firmware have been addressed (PAL video works). 

Here is my config file. Please note that I am using this together with dRonin, so you need to change the CONTROLLER SOFTWARE to whatever you are using.

[config.txt](https://github.com/ShikOfTheRa/scarab-osd/files/4897504/config.txt)

**Comment by ShikOfTheRa** [comment]
@ raemin - many thanks for your help. I am quite sad that dronin development stopped - thought it was going in great direction.

@ unitware - Hello - I had some time to look at this today as much as I can without the hardware. I am not sure what version you are using, but I tried the latest and the vtx menu is missing for me. Once enabled, the screen display looked OK

Very latest for flashing via arduino: 
https://github.com/ShikOfTheRa/scarab-osd/archive/master.zip.

Here is what I see using version above:
![vtxmenu](https://user-images.githubusercontent.com/11336532/87056006-ef5d2600-c1fc-11ea-8bc5-862bc8a80683.png)

**Comment by unitware** [comment]
Thanks for looking into it, I was using the latest on master two days ago. 

I‘m out of office for a week but will look into it again when I’m back. 

Again, thanks for looking at this old hw :) (I think I’ve got a clone since it was discontinued when I bought it so I doubt I will get any help from furious)


Skickat från min iPhone

**Comment by unitware** [comment]
Thanks a lot! 6624262b95e2bc23a55ee1335cc5c12e91f5a53c fixed my remaining issues and I can now select vtx power, band and channel from the osd-menu. The voltage issues also disappeared, maybe because of averaging that is now set to 4.

**Comment by ShikOfTheRa** [comment]
Awesome. I am a little surprised because the menu was completely disabled. I only re-added. But it is great news.

The voltage - that was an error on my part. I was testing a bug I noticed previously and unfortunately left it enabled. It is now fixed!
